ndcube/ndcube.py

```diff
@@ -355,21 +335,45 @@ def axis_world_coords(self, *axes, edges=False):
                           for k in dependent_axes[i]], indexing="ij")
                     for k, axis in enumerate(dependent_axes[i]):
                         quantity_list[axis] = dependent_pixel_quantities[k]
+
                 # Perform wcs translation
                 dependent_axes_coords = self.pixel_to_world(*quantity_list)
+
+                # Wrap the dependent_axes_coords into a list if a single quantity
+                dependent_axes_coords = dependent_axes_coords if isinstance(dependent_axes_coords, list) else [dependent_axes_coords]
                 # Place world coords into output list
                 for dependent_axis in dependent_axes[i]:
                     if dependent_axis in int_axes:
                         # Due to error check above we know dependent
                         # axis can appear in int_axes at most once.
                         j = unique_data_axis(self.wcs, dependent_axis)[0]
                         # Since the dependent_axes_coords contains reduced number of results, adjust the index
-                        axes_coords[j] = dependent_axes_coords[j]
+                        # If dependent_axes_coords is a SkyCoord, then return the SkyCoord, no need to slice
+                        if isinstance(dependent_axes_coords[j], SkyCoord):
+                            # Check if skycoord is True. If False then store the lat/lon
+                            # in axes_coords and insert the other lat/lon
+                            if(skycoord==True):
+                                axes_coords[j] = dependent_axes_coords[j]
+                            else:
+                                k = dependent_axis - j
+                                if(k==0):
+                                    TS = dependent_axes_coords[j].data.lon.deg
+                                elif(k==1):
+                                    TS = dependent_axes_coords[j].data.lat.deg
+
+                                index_of_skcoord = j+1
+                                axes_coords[j] = dependent_axes_coords[j].data.lat.deg if k==0 else dependent_axes_coords[j].data.lon.deg
```

Comment:
Do we want to return values here? Don't we normally return quantities from this method?

```suggestion
                                axes_coords[j] = dependent_axes_coords[j].data.lat if k==0 else dependent_axes_coords[j].data.lon
```

Reply:
This method should be used internally, and many times I had to convert it into `deg` in plotting, so that's the point of it.